bucket/zookeeper.json

```diff
@@ -0,0 +1,47 @@
+{
+    "homepage": "https://zookeeper.apache.org/",
+    "description": "High-performance coordination service for distributed applications",
+    "version": "3.4.13",
+    "license": "Apache-2.0",
+    "url": "https://www.apache.org/dyn/closer.cgi?action=download&filename=zookeeper/zookeeper-3.4.13/zookeeper-3.4.13.tar.gz#/zookeeper-3.4.13.tar.gz",
+    "hash": "sha1:a989b527f3f990d471e6d47ee410e57d8be7620b",
+    "extract_dir": "zookeeper-3.4.13",
+    "bin": [
+        [
+            "bin\\zkServer.cmd",
+            "zkServer"
+        ],
+        [
+            "bin\\zkCli.cmd",
+            "zkCli"
+        ]
+    ],
+    "persist": [
+        "data"
+    ],
+    "suggest": {
+        "JDK": [
+            "java/oraclejdk",
+            "java/oraclejdk8",
+            "java/openjdk"
+        ]
+    },
+    "post_install": [
+        "Copy-Item \"$dir/conf/zoo_sample.cfg\" -Destination \"$dir/conf/zoo.cfg\"",
+        "$dataDirPath = (\"$persist_dir\\data\").replace('\\', '\\\\')",
+        "(Get-Content \"$dir/conf/zoo.cfg\") -Replace '^dataDir=(.+)$', \"dataDir=$dataDirPath\" | Set-Content \"$dir/conf/zoo.cfg\""
+    ],
+    "checkver": {
+        "url": "https://www.apache.org/dist/zookeeper/",
+        "re": "zookeeper-([\\d.]+)/",
+        "reverse": true
+    },
+    "autoupdate": {
+        "url": "https://www.apache.org/dyn/closer.cgi?action=download&filename=zookeeper/zookeeper-$version/zookeeper-$version.tar.gz#/zookeeper-$version.tar.gz",
+        "hash": {
+            "url": "https://www.apache.org/dist/zookeeper/zookeeper-$version/zookeeper-$version.tar.gz.sha1",
+            "find": "(\\w+) "
```

Comment:
Find could be removed. It would be automatically fetched in file.

Reply:
You sure? I simulated an update and it falls back to downloading the file to compute the checksum.

Exact steps to reproduce:
1. Remove the `find` command as you propose
1. Lower the version number to `3.4.12` in the app manifest
1. Execute `.\bin\checkver.ps1 zookeeper -u`

> Searching hash for closer.cgi in https://www.apache.org/dist/zookeeper/zookeeper-3.4.13/zookeeper-3.4.13.tar.gz.sha1
Could not find hash in https://www.apache.org/dist/zookeeper/zookeeper-3.4.13/zookeeper-3.4.13.tar.gz.sha1

Hash file at https://www.apache.org/dist/zookeeper/zookeeper-3.4.13/zookeeper-3.4.13.tar.gz.sha1 looks like this:
> a989b527f3f990d471e6d47ee410e57d8be7620b  zookeeper-3.4.13.tar.gz

I guess the problem is that it looks for a hash for `closer.cgi` when it should be looking for `zookeeper-3.4.13.tar.gz`.

Reply:
You are right. I misslooked that parameters.
Change it to:
```json
        "url": "https://www.apache.org/dist/zookeeper/zookeeper-$version/zookeeper-$version.tar.gz",
        "hash": {
            "url": "$url.sha1"
        },
```

Reply:
![Proof](https://i.imgur.com/8OPgg4k.png)

Reply:
Also. Use `-f` parameter for this updates. No need to change version to force update.